src/api/RequestApi.ts

```diff
@@ -1,10 +1,55 @@
-import { spWebContext } from "../providers/SPWebContext";
 import { ApiError } from "./InternalErrors";
 import { IItemUpdateResult } from "@pnp/sp/items";
 import { EMPTYPES } from "../constants/EmpTypes";
 import { worklocation } from "../constants/WorkLocations";
 import { SPPersona } from "../components/PeoplePicker/PeoplePicker";
 
+import { spWebContext } from "../providers/SPWebContext";
+import { useQuery } from "@tanstack/react-query";
+
+export interface IRequest {
+  Id: number;
+  empName: string;
+  eta: string | null;
+  completionDate: string | null;
+}
+
+const getMyRequests = (userId: number | undefined) => {
+  if (process.env.NODE_ENV === "development") {
+    const date = new Date().toISOString();
+    return Promise.resolve([
+      {
+        Id: 1,
+        empName: "Doe, John D",
+        eta: date,
+        completionDate: date,
+      },
+      {
+        Id: 2,
+        empName: "Doe, Jane D",
+        eta: date,
+        completionDate: date,
+      },
+    ] as IRequest[]);
+  } else if (userId === undefined) {
+    console.log("No user specified");
+    return Promise.reject([] as IRequest[]);
+  } else {
+    return spWebContext.web.lists
+      .getByTitle("requests")
+      .items.filter(
+        `supGovLead/Id eq '${userId}' or employee/Id eq '${userId}'`
+      )();
+  }
+};
```

Comment:
This should be part of the API, right?  And have 2 different versions based on the environment.  You can leverage the existing data records defined in test, and add some extras if desired.

Reply:
Perfect!  My main concern was embedding test data in each of the functions, instead of using a single test bank.  I see us turning your date conversions into a function we call by each of the hooks to transform the data.  That is one of the issues standing with converting to SP now, is that the date objects are actually strings under current "production" API.